bucket_policy.tf

```diff
@@ -13,6 +13,119 @@ resource "aws_s3_bucket_policy" "bucket" {
 */
 data "aws_iam_policy_document" "bucket" {
 
+  dynamic "statement" {
+    for_each = length(local.metadata_read_services) > 0 ? [1] : []
+
+    content {
+      sid    = "AllowReadBucketMetadataAWSServices"
```

Comment:
`Allow....` but the `effect` is `"Deny"`. The `sid` doesn't immediately seem aligned with the `effect` which makes it tricky to think through the statement.